docs/docs/reference/built-in-components/gatsby-slice.md

```diff
@@ -0,0 +1,164 @@
+
+```typescript:title=gatsby-node.ts
+exports.createPages = async ({ actions }) => {
+  actions.createSlice({
+    id: `navigation-bar`,
+    component: require.resolve(`./src/components/navigation-bar.js`),
+  })
+
+  // ...
+}
+```
+
+Now that we have a `navigation-bar` Slice created, let's go use it!
+
+### Using created Slices
+
+Now we need to convert the `DefaultLayout` component to actually use the new Slice we created.
+
+```diff
+-import { NavigationBar, Footer } from "../components"
++import { Footer } from "../components"
++import { Slice } from "gatsby"
+
+export const DefaultLayout = ({ children, navigationBarClassName }: DefaultLayoutProps) => {
+  /// ...
+
+  return (
+    <div className={styles.defaultLayout} />
+-     <NavigationBar className={navigationBarClassName} />
++     <Slice alias="navigation-bar" className={navigationBarClassName} />
+      {content}
+      <Footer />
+    </div>
+  )
+}
+```
+
+That's it! After a successful `yarn gatsby build`, you should see a list of `Slices` that were built for your site.
+
+### Using Context
+
+Similar to the context that can be passed to pages in [`createPages`](/docs/reference/config-files/gatsby-node/#createPages), [`createSlice`](/docs/reference/config-files/actions/#createNodeField) can also pass context to individual slices.
+
+```typescript:title=gatsby-node.ts
+exports.createPages = async ({ actions }) => {
+  actions.createSlice({
+    id: `navigation-bar`,
+    context: {
+      jokeOfTheDay: `What's blue and not heavy? Light blue.`,
+    }
+    component: require.resolve(`./src/components/navigation-bar.js`),
+  })
+
+  // ...
+}
+```
+
+The data passed to `context` here will be handed down to the `NavigationBar` component with the key `sliceContext`.
+
+```typescript:title=src/components/navigation-bar.tsx
+// highlight-next-line
+export const NavigationBar = ({ className, sliceContext }: NavigationBarProps) => {
+  // ...
+
+  return (
+    <div className={className}>
+      <Link to="/">Home</Link>
+      // highlight-next-line
+      <Link to="/jokes">{sliceContext.jokeOfTheDay}</Link>
+    </div>
+  )
+}
+```
+
+### Using Aliases
+
+There will be times where a single Slice will either need to be handed different context or swapped entirely depending on which page it's being rendered on.
+
+This is where we can utilize the `alias` prop that is given to the `<Slice>` component. When we converted the `<NavigationBar>` component to a slice, we created the slice with an `id` of `navigation-bar`, but passed that value to the `alias` prop of `<Slice>`. Why is there a difference in key names?
+
+An `alias` is not a 1-to-1 mapping of string-to-slice. When we create a page using [`createPages`](/docs/reference/config-files/gatsby-node/#createPages), we can actually pass a key-value map of alias-to-id to tell Gatsby which Slice to use throughout the page.
```

Comment:
These two sentences have too much jargon for my taste (https://www.gatsbyjs.com/contributing/gatsby-style-guide/#define-jargon) and I had difficulties understanding it